In [3]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

# Load the dataset
data = pd.read_csv('/workspaces/codespaces-jupyter/data/Copy of loan.csv')
data.shape

(614, 13)

In [4]:
data.info()

data['Gender'] = data['Gender'].replace('Male', 1).replace(['Female', 'Other'], 0)



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(5), int64(1), object(7)
memory usage: 62.5+ KB


In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    float64
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(6), int64(1), object(6)
memory usage: 62.5+ KB


In [6]:
data['Loan_Status'] = data['Loan_Status'].replace('Y', 1).replace(['N'], 0)
loan_status_items = data['Loan_Status'].unique()

print(loan_status_items)


[0 1]


In [9]:
data['Education'] = data['Education'].replace('Graduate', 1).replace(['Not Graduate'], 0)
education_status_items = data['Education'].unique()

print(education_status_items)



[0 1]


In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    float64
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    int64  
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    int64  
dtypes: float64(6), int64(3), object(4)
memory usage: 62.5+ KB


In [14]:
data = data.dropna()

In [15]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error


# Select the columns you want to use as features
features = ['Gender', 'Dependents', 'Education', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History']

# Select the column you want to predict
target = 'Loan_Status'

# Split the data into training and test sets
X_train, X_test, y_train, y_test = train_test_split(data[features], data[target], test_size=0.2)

# Create and fit the model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions on the test set
y_pred = model.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
print(f'Mean squared error: {mse:.2f}')

# Show the coefficients for each feature
coefficients = pd.DataFrame({'Feature': features, 'Coefficient': model.coef_})
print(coefficients)

Mean squared error: 0.18
             Feature   Coefficient
0             Gender  5.841470e-02
1         Dependents  2.659132e-02
2          Education  3.157980e-02
3    ApplicantIncome -6.417009e-07
4  CoapplicantIncome -1.384382e-05
5         LoanAmount -2.242299e-04
6   Loan_Amount_Term  7.754130e-05
7     Credit_History  7.144768e-01


The output you've provided is the result of running a linear regression model on a dataset. The model is predicting the 'Loan_Status' based on various features like 'Gender', 'Dependents', 'Education', etc.

The 'Mean squared error' is a measure of how well the model is performing, with lower values indicating better performance. In this case, the MSE is 0.18.

The table below the MSE is showing the coefficients of each feature in the linear regression model. These coefficients represent the relationship between each feature and the target variable 'Loan_Status'. For example, the coefficient for 'Gender' is 5.841470e-02, which means for each unit increase in 'Gender', the 'Loan_Status' increases by 5.841470e-02 units, holding all other features constant.

However, it seems like you're asking for an explanation of Markdown in the context of this Python code. Markdown is not directly relevant to this Python code. Markdown is a lightweight markup language for creating formatted text. It's often used for writing rich text on the web and in text editors. In a Jupyter notebook, you can use Markdown to write explanatory text, add headings, create lists, and so on. But in the Python code you've provided, there's no Markdown being used.